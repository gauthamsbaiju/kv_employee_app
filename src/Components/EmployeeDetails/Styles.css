.details-wrapper{
    width: calc(100% * (1/5) - 10px - 1px);
    
}

.details-label{
    padding-bottom: 10px;
}