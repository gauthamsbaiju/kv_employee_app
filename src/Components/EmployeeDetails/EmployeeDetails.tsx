import React from 'react';
import './Styles.css';
import Status from '../Status/Status';

export type EmpPropsType = {
  lable: string;
  value: string | number | boolean;
  type: string;
};

const EmployeeDetails: React.FC<EmpPropsType> = (props) => {
  return (
    <div className='details-wrapper'>
      <div className='details-label' data-testid='employeeDetailsLabel-test'>
        {props.lable}
      </div>
      {props.type === 'status' && <Status value={Boolean(props.value)} />}
      <div data-testid='employeeDetailsValue-test'>{props.value}</div>
    </div>
  );
};

export default EmployeeDetails;
