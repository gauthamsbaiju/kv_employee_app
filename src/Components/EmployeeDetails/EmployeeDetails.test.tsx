import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import EmployeeDetails, { EmpPropsType } from './EmployeeDetails';

describe('Test for employee details component', () => {
  test('Test for label rendered', () => {
    const EmpProps: EmpPropsType = {
      lable: 'Name',
      value: 'Abc',
      type: 'text'
    };

    render(<EmployeeDetails {...EmpProps} />);
    const element = screen.getByTestId('employeeDetailsLabel-test');

    if (element) expect(element).toHaveTextContent('Name');
  });

  test('Test for snapshot of label rendered', () => {
    const EmpProps: EmpPropsType = {
      lable: 'Name',
      value: 'Abc',
      type: 'text'
    };

    render(<EmployeeDetails {...EmpProps} />);
    const element = screen.getByTestId('employeeDetailsLabel-test');

    expect(element).toMatchSnapshot();
  });

  test('Test for snapshot of data rendered', () => {
    const EmpProps: EmpPropsType = {
      lable: 'Name',
      value: 'Abc',
      type: 'text'
    };

    render(<EmployeeDetails {...EmpProps} />);
    const element = screen.getByTestId('employeeDetailsValue-test');

    expect(element).toMatchSnapshot();
  });

  test('Test for value of employee detail rendered', () => {
    const EmpProps: EmpPropsType = {
      lable: 'Name',
      value: 'Abc',
      type: 'text'
    };

    render(<EmployeeDetails {...EmpProps} />);
    const element = screen.getByTestId('employeeDetailsValue-test');

    expect(element).toHaveTextContent('Abc');
  });
});
