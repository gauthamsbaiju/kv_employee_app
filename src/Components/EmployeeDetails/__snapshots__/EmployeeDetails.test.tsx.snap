// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for employee details component Test for snapshot of data rendered 1`] = `
<div
  data-testid="employeeDetailsValue-test"
>
  Abc
</div>
`;

exports[`Test for employee details component Test for snapshot of label rendered 1`] = `
<div
  class="details-label"
  data-testid="employeeDetailsLabel-test"
>
  Name
</div>
`;
